upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template

```diff
@@ -639,286 +694,295 @@ data:
+subjects:
+  - kind: ServiceAccount
+    name: karpenter
+    namespace: kube-system
 ---
-# Source: karpenter/templates/webhook/rbac.yaml
+# Source: karpenter/templates/rolebinding.yaml
 apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
-  name: karpenter-webhook
+  name: karpenter-dns
   namespace: kube-system
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
-  name: karpenter-webhook
+  name: karpenter-dns
 subjects:
-- kind: ServiceAccount
-  name: karpenter
-  namespace: kube-system
+  - kind: ServiceAccount
+    name: karpenter
+    namespace: kube-system
 ---
-# Source: karpenter/templates/controller/deployment.yaml
+# Source: karpenter/templates/service.yaml
 apiVersion: v1
 kind: Service
 metadata:
-  name: karpenter-metrics
+  name: karpenter
   namespace: kube-system
 spec:
+  type: ClusterIP
   ports:
-    - port: 8080
+    - name: http-metrics
+      port: 8080
       targetPort: http-metrics
-  selector:
-    karpenter: controller
----
-# Source: karpenter/templates/webhook/deployment.yaml
-apiVersion: v1
-kind: Service
-metadata:
-  name: karpenter-webhook
-  namespace: kube-system
-spec:
-  ports:
-    - port: 443
+      protocol: TCP
+    - name: https-webhook
+      port: 443
       targetPort: https-webhook
+      protocol: TCP
   selector:
-    karpenter: webhook
+    app.kubernetes.io/name: karpenter
+    app.kubernetes.io/instance: karpenter
+
+
+
 ---
-# Source: karpenter/templates/controller/deployment.yaml
+# Source: karpenter/templates/deployment.yaml
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: karpenter-controller
+  name: karpenter
   namespace: kube-system
 spec:
   replicas: {{ ControlPlaneControllerReplicas false }}
+  revisionHistoryLimit: 10
   strategy:
-    type: Recreate
+    rollingUpdate:
+      maxUnavailable: 1
   selector:
     matchLabels:
-      karpenter: controller
+      app.kubernetes.io/name: karpenter
+      app.kubernetes.io/instance: karpenter
   template:
     metadata:
       labels:
-        karpenter: controller
         app.kubernetes.io/name: karpenter
+        app.kubernetes.io/instance: karpenter
     spec:
-      affinity:
-        nodeAffinity:
-          requiredDuringSchedulingIgnoredDuringExecution:
-            nodeSelectorTerms:
-            - matchExpressions:
-              - key: node-role.kubernetes.io/control-plane
-                operator: Exists
-            - matchExpressions:
-              - key: node-role.kubernetes.io/master
-                operator: Exists
-      priorityClassName: system-cluster-critical
       serviceAccountName: karpenter
+      securityContext:
+        fsGroup: 1000
+      priorityClassName: "system-cluster-critical"
       dnsPolicy: Default
       containers:
         - name: controller
-          image: public.ecr.aws/karpenter/controller:v0.16.3
+          image: public.ecr.aws/karpenter/controller:v0.24.0@sha256:83e7a21200efd4e17314c931aa715c9c73b27fcbf08673d263c007029ad754da
```

Comment:
Please don't include these sha. kOps will automatically add the SHAs for all manifests during cloudup. This allows a bit more flexibility for those who want to rehost/rebuild images.

Reply:
Done :pray: 
 - https://github.com/kubernetes/kops/pull/15144/commits/213d11cfb8aa083e6930094e452229b62d1b02b4